nkms/characters.py

```diff
@@ -281,9 +285,32 @@ def learn_about_nodes(self, address, port):
         """
         Sends a request to node_url to find out about known nodes.
         """
-        # TODO: Find out about other known nodes, not just this one.  #175
-        node = Ursula.from_rest_url(address, port)
-        self.known_nodes[node.interface_dht_key()] = node
+        response = self.network_middleware.get_nodes_via_rest(address, port)
+        signature, nodes = signature_splitter(response.content, return_remainder=True)
+        # TODO: Although not treasure map-related, this has a whiff of #172.
+        ursula_interface_splitter = dht_value_splitter + BytestringSplitter((bytes, 17))
+        split_nodes = ursula_interface_splitter.repeat(nodes)
+        new_nodes = {}
+        for node_meta in split_nodes:
+            header, sig, pubkey, interface_info = node_meta
+            if not pubkey in self.known_nodes:
+                if sig.verify(keccak_digest(interface_info), pubkey):
+                    address, dht_port, rest_port = msgpack.loads(interface_info)
+                    new_nodes[pubkey] = \
+                        Ursula.as_discovered_on_network(
+                            rest_port=rest_port,
+                            dht_port=dht_port,
+                            ip_address=address.decode("utf-8"),
+                            powers_and_keys=({SigningPower: pubkey})
+                        )
+                else:
+                    self.log.warn("Discovered node with bad signature: {}".format(node_meta))
```

Comment:
This is kinda a big deal. We only want to warn about this?

Reply:
Yeah, I think so.  Although it's a big deal in the sense that we've seemingly found an Ursula who is doing the Wrong Thing(TM), it's not a big deal to this individual Alice/Bob.  They will almost certainly be able to continue to transmit, so I think they'd be pretty annoyed if this raised instead.

Hopefully we get some valuable reports in this scenario.

Reply:
I agree – if I've understood correctly, this would be issue for the network to deal with (possibly through a challenge), not the customer. In the event of a bad signature, would that node be removed from the group that is performing the re-encryption, and replaced by another?

Reply:
Well, in this case, the user is just learning about any available nodes on the network.  These aren't necessarily the nodes that will be performing re-encryption later.  If the node being learned about doesn't have a valid signature for its ID, then we don't even consider it real - we just toss it aside and emit a warning.

So, @arjunhassard, no, the node isn't remove per se.  Hopefully, a different Ursula provides us with a valid signature for the node's ID.

At the end of the day, the node *from whom we're learning* has done the wrong thing here - they were supposed to refrain from propagating this information after they themselves checked the Signature.  

I have modified the warning message to reflect this in #246, the PR that follows this one.